api/observability/v1/clusterlogforwarder_types.go

```diff
@@ -0,0 +1,190 @@
+	//
+	// +required
+	OutputRefs []string `json:"outputRefs,omitempty"`
+
+	// Filters lists the names of filters to be applied to records going through this pipeline.
+	//
+	// Each filter is applied in order.
+	// If a filter drops a records, subsequent filters are not applied.
+	// +optional
+	FilterRefs []string `json:"filterRefs,omitempty"`
+}
+
+type ServiceAccountSpec struct {
+	// Name is the serviceaccount associated with the clusterlogforwarder
+	// +required
+	Name string `json:"name,omitempty"`
+}
+
+type LimitSpec struct {
+	// MaxRecordsPerSecond is the maximum number of log records
+	// allowed per input/output in a pipeline
+	//
+	// +required
+	MaxRecordsPerSecond int64 `json:"maxRecordsPerSecond,omitempty"`
+}
+
+// ResourceTypeSpec is the spec for defining a key/value resource type
+type ResourceTypeSpec struct {
+	// +kubebuilder:validation:Enum:=secret,configmap
+	// kubebuilder:default:=secret
+	// +required
+	Type string `json:"type,omitempty"`
+
+	// Name of the given resource type
+	//
+	// +kubebuilder:validation:minLength:=1
+	// +required
+	Name string `json:"name,omitempty"`
+}
+
+type KeySpec struct {
+	// The key name found in the resource
+	//
+	// +kubebuilder:validation:minLength:=1
+	// +required
+	key string `json:"key,omitempty"`
+}
+
+// PriorityKeySpec spec a key from a specific key/value resource type that takes precedence over any defaults
+type PriorityKeySpec struct {
+	ResourceTypeSpec `json:",inline"`
+
+	KeySpec `json:",inline"`
+}
+
+// ClusterLogForwarderStatus defines the observed state of ClusterLogForwarder
+type ClusterLogForwarderStatus struct {
+	// Conditions of the log forwarder.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Forwarder Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:forwarderConditions"}
+	Conditions status.Conditions `json:"conditions,omitempty"`
+
+	// Inputs maps input name to condition of the input.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Input Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:inputConditions"}
+	Inputs NamedConditions `json:"inputs,omitempty"`
+
+	// Outputs maps output name to condition of the output.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Output Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:outputConditions"}
+	Outputs NamedConditions `json:"outputs,omitempty"`
+
+	// Filters maps filter name to condition of the filter.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Filter Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:filterConditions"}
+	Filters NamedConditions `json:"filters,omitempty"`
+
+	// Pipelines maps pipeline name to condition of the pipeline.
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Pipeline Conditions",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:pipelineConditions"}
+	Pipelines NamedConditions `json:"pipelines,omitempty"`
```

Comment:
The `xDescriptors` seem not to be correct here. AFAIU the only allowed descriptors are given here: https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md